modules/apps/style-book/style-book-web/src/main/java/com/liferay/style/book/web/internal/portlet/action/ViewMVCRenderCommand.java

```diff
@@ -29,7 +39,43 @@ public class ViewMVCRenderCommand implements MVCRenderCommand {
 	public String render(
 		RenderRequest renderRequest, RenderResponse renderResponse) {
 
+		HttpServletRequest httpServletRequest =
+			_portalUtil.getHttpServletRequest(renderRequest);
+		LiferayPortletRequest liferayPortletRequest =
+			_portalUtil.getLiferayPortletRequest(renderRequest);
+		LiferayPortletResponse liferayPortletResponse =
+			_portalUtil.getLiferayPortletResponse(renderResponse);
+
+		StyleBookDisplayContext styleBookDisplayContext =
+			new StyleBookDisplayContext(
+				httpServletRequest, liferayPortletRequest,
+				liferayPortletResponse);
+
+		renderRequest.setAttribute(
+			StyleBookDisplayContext.class.getName(), styleBookDisplayContext);
+
+		StyleBookManagementToolbarDisplayContext
+			styleBookManagementToolbarDisplayContext =
+				new StyleBookManagementToolbarDisplayContext(
+					_cetManager, httpServletRequest, liferayPortletRequest,
+					liferayPortletResponse, _frontendTokenDefinitionRegistry,
+					styleBookDisplayContext.
+						getStyleBookEntriesSearchContainer());
+
+		renderRequest.setAttribute(
+			StyleBookManagementToolbarDisplayContext.class.getName(),
+			styleBookManagementToolbarDisplayContext);
+
 		return "/view.jsp";
 	}
 
+	@Reference
+	private CETManager _cetManager;
+
+	@Reference
+	private FrontendTokenDefinitionRegistry _frontendTokenDefinitionRegistry;
+
+	@Reference
+	private PortalUtil _portalUtil;
```

Comment:
I think  we can replace this for `"private Portal _portal;"`. I recall that I read about use injected service(`Portal`, in this case) instead of a Utility class :thinking:  , though I dont remember where. cc: @thiago-buarqque @ethib137

Reply:
Yes, good one @algobob , I missed it